omc3/scripts/kmod_summary.py

```diff
@@ -76,144 +99,235 @@ def output_kmod_summary_tables(
+    """
+    LOG.info("Grouping averaged kmod results.")
+
+    grouped_averaged = []
+    for av_ip, av_res in averaged_meas.items():
+        LOG.info(f"Reading averaged results: {BEAM_DIR}{beam}, {av_ip}")
+        av_tab = av_res[0] # 0: averaged results table, 1, 2: are betx, bety for IP closest elements
+        av_tab_beam = av_tab.loc[[beam]]
+        df_averaged = av_tab_beam[COLS_X + COLS_Y]
+        df_averaged.insert(0, IP_COLUMN, av_ip)
+        grouped_averaged.append(df_averaged)
+    return grouped_averaged
+
+def _get_lumi_imbalance(
+    output_dir:Path | str
+    ) -> str:
+
+    """
+    Gathers the various luminosity imbalance dataframes in a single str, one line for each file.
+
+    Args:
+        output_dir (Path | str ): Path to the lumi imbalance files.
+    Returns:
+        str: str containing grouped luminosity imbalance results.
+    """
+    LOG.info("Getting luminosity imbalance values.")
+    output_dir = Path(output_dir)
+    prefix = EFFECTIVE_BETAS_FILENAME.split("{")[0] # if the file name starts with EFFECTIVE_BETAS_FILENAME till {ip}
+
+    report_lines = []
+    for file_path in output_dir.iterdir():
+        if not file_path.is_file() or not file_path.name.startswith(prefix):
+            continue
+
+        df = tfs.read(file_path)
+        lumi_imbalance = float(df.headers.get("LUMIIMBALANCE", 0))
+        err_lumi_imbalance = float(df.headers.get("ERRLUMIIMBALANCE", 0))
+        names = df["NAME"].tolist()
+
+        # final check
+        if lumi_imbalance is not None and err_lumi_imbalance is not None and len(names) >= 2:
+            report_lines.append(f"Luminosity imbalance in between {names[0]} and {names[1]} is {lumi_imbalance} ± {err_lumi_imbalance}")
+
+    return "\n".join(report_lines)
 
-    return kmod_summaries, tables_logbook
+def _prepare_logbook_table(
+    beam: int,
+    kmod_summary: tfs.TfsDataFrame,
+    kmod_summary_averaged: tfs.TfsDataFrame | None = None,
+    kmod_summary_lumiimb: str | None = None,
+    ) -> list[str]:
+
+    """
+    Prepare formatted logbook tables from K-modulation summary data.
+
+    Args:
+        beam (int): Beam number to process.
+        kmod_summary (tfs.TfsDataFrame): DataFrame containing collected Kmod results.
+        kmod_summary_averaged (tfs.TfsDataFrame | None): Optional DataFrame containing averaged Kmod results.
+        kmod_summary_lumiimb (str | None): Optional txt file containing luminosity imbalance results.
+
+    Returns:
+        list[str]: List of formatted string tables. Includes separate sections for X-plane, Y-plane, and optionally averaged results.
+    """
+    LOG.info("Creating .txt tables.")
+
+    kmod_summary_x = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_X]
+    kmod_summary_y = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_Y]
+    kmod_summary_x_averaged = kmod_summary_averaged[[IP_COLUMN] + COLS_X]
+    kmod_summary_y_averaged = kmod_summary_averaged[[IP_COLUMN] + COLS_Y]
```

Comment:
You need to handle the case where `kmod_summary_averaged` is not given. The checks below look at `kmod_summary_[xy]_averaged` being `None` or not, but you didn't check here.
```suggestion
    kmod_summary_x: tfs.TfsDataFrame = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_X]
    kmod_summary_y: tfs.TfsDataFrame = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_Y]
    kmod_summary_x_averaged: tfs.TfsDataFrame | None = kmod_summary_averaged[[IP_COLUMN] + COLS_X] if kmod_summary_averaged is not None else None
    kmod_summary_y_averaged: tfs.TfsDataFrame | None = kmod_summary_averaged[[IP_COLUMN] + COLS_Y] if kmod_summary_averaged is not None else None
```